src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupService.scala

```diff
@@ -37,53 +39,53 @@ class ManagedGroupService(resourceService: ResourceService, val resourceTypes: M
     val email = generateManagedGroupEmail(resource.resourceId)
     val workbenchGroupName = WorkbenchGroupName(resource.resourceId.value)
     val groupMembers: Set[WorkbenchSubject] = componentPolicies.map(_.id)
-    resourceService.directoryDAO.createGroup(BasicWorkbenchGroup(workbenchGroupName, groupMembers, email))
+    directoryDAO.createGroup(BasicWorkbenchGroup(workbenchGroupName, groupMembers, email))
   }
 
   private def generateManagedGroupEmail(resourceId: ResourceId): WorkbenchEmail = {
-    val localPart = resourceId.value
-    validateEmail(localPart)
-    WorkbenchEmail(constructEmail(localPart))
+    val groupName = resourceId.value
+    validateGroupName(groupName)
+    WorkbenchEmail(constructEmail(groupName))
   }
 
-  private def constructEmail(localPart: String) = {
-    s"${localPart}@${resourceService.emailDomain}"
+  private def constructEmail(groupName: String) = {
+    s"${groupName}@${emailDomain}"
   }
 
-  private def validateEmail(localPart: String) = {
-    val errors = validateLocalPartPattern(localPart) ++ validateEmailLength(constructEmail(localPart))
+  private def validateGroupName(groupName: String) = {
+    val errors = validateGroupNamePattern(groupName) ++ validateGroupNameLength(constructEmail(groupName))
     if (errors.nonEmpty)
-      throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "Cannot create valid email address from group name" , errors.toSeq))
+      throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "Cannot create valid email address with the group name provided" , errors.toSeq))
```

Comment:
Minor improvement: include the group name in the error message.

Reply:
fixed.